crates/goose-server/src/routes/agent.rs

```diff
@@ -42,23 +55,144 @@ pub struct AddSubRecipesResponse {
 pub struct UpdateProviderRequest {
     provider: String,
     model: Option<String>,
+    #[allow(dead_code)]
+    session_id: String,
 }
 
 #[derive(Deserialize, utoipa::ToSchema)]
 pub struct SessionConfigRequest {
     response: Option<Response>,
+    #[allow(dead_code)]
+    session_id: String,
 }
 
 #[derive(Deserialize, utoipa::ToSchema)]
 pub struct GetToolsQuery {
     extension_name: Option<String>,
+    #[allow(dead_code)]
+    session_id: String,
+}
+
+#[derive(Deserialize, utoipa::ToSchema)]
+pub struct UpdateRouterToolSelectorRequest {
+    #[allow(dead_code)]
+    session_id: String,
+}
+
+#[derive(Deserialize, utoipa::ToSchema)]
+pub struct StartAgentRequest {
+    working_dir: String,
+}
+
+#[derive(Deserialize, utoipa::ToSchema)]
+pub struct ResumeAgentRequest {
+    session_id: String,
+}
+
+// This is the same as SessionHistoryResponse
+#[derive(Serialize, utoipa::ToSchema)]
+pub struct StartAgentResponse {
+    session_id: String,
+    metadata: SessionMetadata,
+    messages: Vec<Message>,
 }
 
 #[derive(Serialize, utoipa::ToSchema)]
 pub struct ErrorResponse {
     error: String,
 }
 
+#[utoipa::path(
+    post,
+    path = "/agent/start",
+    request_body = StartAgentRequest,
+    responses(
+        (status = 200, description = "Agent started successfully", body = StartAgentResponse),
+        (status = 400, description = "Bad request - invalid working directory"),
+        (status = 401, description = "Unauthorized - invalid secret key"),
+        (status = 500, description = "Internal server error")
+    )
+)]
+async fn start_agent(
+    State(state): State<Arc<AppState>>,
+    headers: HeaderMap,
+    Json(payload): Json<StartAgentRequest>,
+) -> Result<Json<StartAgentResponse>, (StatusCode, Json<ErrorResponse>)> {
+    verify_secret_key(&headers, &state).map_err(|_| {
+        (
+            StatusCode::UNAUTHORIZED,
+            Json(ErrorResponse {
+                error: "Unauthorized - Invalid or missing API key".to_string(),
+            }),
+        )
+    })?;
+
+    let session_id = session::generate_session_id();
+    let counter = state.session_counter.fetch_add(1, Ordering::SeqCst) + 1;
+
+    let metadata = SessionMetadata {
```

Comment:
does SessionMetadata::new() not do what we want?

Reply:
well, I added a description with a counter to this so we can better see which session is new etc. but you are right, I should just move that to the new method